### PR TITLE
Storage - code generator update

### PR DESCRIPTION
**Pull request by tanyasethi-msft**
- [x] The purpose of this PR is explained in this or a referenced issue.
- [] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to module CHANGELOG.md are included.
- [x] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go

### CODE REVIEW

File: sdk/storage/azblob/internal/generated/zz_appendblob_client.go [hunk 1]
```diff
@@ -72,54 +69,54 @@ func (client *AppendBlobClient) appendBlockCreateRequest(ctx context.Context, co
...
 	}
 	req.Raw().URL.RawQuery = reqQP.Encode()
+	req.Raw().Header["Accept"] = []string{"application/xml"}
 	req.Raw().Header["Content-Length"] = []string{strconv.FormatInt(contentLength, 10)}
 	if options != nil && options.TransactionalContentMD5 != nil {
 		req.Raw().Header["Content-MD5"] = []string{base64.StdEncoding.EncodeToString(options.TransactionalContentMD5)}
 	}
-	if options != nil && options.TransactionalContentCRC64 != nil {
-		req.Raw().Header["x-ms-content-crc64"] = []string{base64.StdEncoding.EncodeToString(options.TransactionalContentCRC64)}
+	if modifiedAccessConditions != nil && modifiedAccessConditions.IfMatch != nil {
```
Comment (vibhansa-msft) [inline sdk/storage/azblob/internal/generated/zz_appendblob_client.go:77]: We have both options and modifiedAccessConditions as input param to this function. Is it duplicate?

File: sdk/storage/azdatalake/internal/generated_blob/zz_options.go [hunk 2]
```diff
@@ -0,0 +1,1463 @@
...
+// Code generated by Microsoft (R) AutoRest Code Generator. DO NOT EDIT.
+// Changes may cause incorrect behavior and will be lost if the code is regenerated.
+
+package generated_blob
+
+import "time"
+
+// AppendBlobClientAppendBlockFromURLOptions contains the optional parameters for the AppendBlobClient.AppendBlockFromURL
+// method.
+type AppendBlobClientAppendBlockFromURLOptions struct {
```
Comment (vibhansa-msft) [inline sdk/storage/azdatalake/internal/generated_blob/zz_options.go:12]: This is a new file being added, which means these options were earlier part of some other file and now being removed from there ?

Comment (tanyasethi-msft) [reply sdk/storage/azdatalake/internal/generated_blob/zz_options.go:12]: Yes, these were part of models.go before, which have been moved to a new file

File: sdk/storage/azblob/internal/generated/zz_appendblob_client.go [hunk 3]
```diff
@@ -1,6 +1,3 @@
-//go:build go1.18
-// +build go1.18
```
Comment (souravgupta-msft) [inline sdk/storage/azblob/internal/generated/zz_appendblob_client.go:2]: The code generator update removed these build statements from all the generated files. Is this expected? @jhendrixMSFT

**Review by souravgupta-msft — APPROVED** [review]

File: sdk/storage/azblob/internal/generated/zz_appendblob_client.go (hunk 3 above)
Comment (jhendrixMSFT) [reply sdk/storage/azblob/internal/generated/zz_appendblob_client.go:2]: Yes. The build constraints are no longer required.

File: sdk/storage/azdatalake/internal/generated/autorest.md [hunk 4]
```diff
@@ -146,7 +146,7 @@ directive:
 
 ``` yaml
 directive:
-- from: zz_models.go
+- from: zz_options.go
```
Comment (souravgupta-msft) [inline sdk/storage/azdatalake/internal/generated/autorest.md:149]: The below transform should be applied to zz_models.go

File: sdk/storage/azdatalake/internal/generated/zz_models.go [hunk 5]
```diff
@@ -52,7 +45,7 @@ type PathPrefix struct {
 // PathPropertiesInternal - Properties of a blob
 type PathPropertiesInternal struct {
 	// REQUIRED
-	ETag *azcore.ETag `xml:"Etag"`
+	Etag *string `xml:"Etag"`
```
Comment (souravgupta-msft) [inline sdk/storage/azdatalake/internal/generated/zz_models.go:48]: This type should be `azcore.ETag`.

File: sdk/storage/azdatalake/internal/generated_blob/zz_options.go [hunk 6]
```diff
@@ -0,0 +1,1463 @@
...
+// LeaseAccessConditions contains a group of parameters for the ContainerClient.GetProperties method.
+type LeaseAccessConditions struct {
+	// If specified, the operation only succeeds if the resource's lease is active and matches this ID.
+	LeaseID *string
+}
+
+// ModifiedAccessConditions contains a group of parameters for the ContainerClient.Delete method.
+type ModifiedAccessConditions struct {
+	// Specify an ETag value to operate only on blobs with a matching value.
+	IfMatch *string
```
Comment (souravgupta-msft) [inline sdk/storage/azdatalake/internal/generated_blob/zz_options.go:1091]: IfMatch and IfNoneMatch should of type `*azcore.ETag`

File: sdk/storage/azdatalake/internal/generated/zz_models.go (hunk 5 above)
Comment (tanyasethi-msft) [reply sdk/storage/azdatalake/internal/generated/zz_models.go]: Thanks! Regenerated this after transform update.

File: sdk/storage/azdatalake/internal/generated_blob/zz_options.go (hunk 6 above)
Comment (tanyasethi-msft) [reply sdk/storage/azdatalake/internal/generated_blob/zz_options.go]: Corrected this, thanks! :)

File: sdk/storage/azdatalake/internal/generated/autorest.md (hunk 4 above)
Comment (tanyasethi-msft) [reply sdk/storage/azdatalake/internal/generated/autorest.md]: This is corrected in the next commits.

**Review by souravgupta-msft — APPROVED** [review]

**Pull request merged**